server/src/main/java/org/opensearch/common/settings/ClusterSettings.java

```diff
@@ -529,6 +529,7 @@ public void apply(Settings value, Settings current, Settings previous) {
                 Node.BREAKER_TYPE_KEY,
                 OperationRouting.USE_ADAPTIVE_REPLICA_SELECTION_SETTING,
                 OperationRouting.IGNORE_AWARENESS_ATTRIBUTES_SETTING,
+                OperationRouting.USE_WEIGHTED_ROUND_ROBIN,
```

Comment:
I agree, this will require us to make two api calls. But the advantage is, it's easy to turn off the feature when needed with just setting ```cluster.routing.use_weighted_round_robin``` to ```false```. The weights are stored in cluster metadata and in case we go just with metadata setting then we need to clear the metadata details from cluster state in case we need to turn off this WRR based shard routing.

Reply:
That should be fine. Turning off feature and turning it on should be similar APIs. Else it can become confusing to the user.

Reply:
discussed with @gbbafna , will remove the cluster setting ```cluster.routing.use_weighted_round_robin``` and add a ```DELETE``` api to clear weights from cluster metadata if we need to turn off WRR shard routing.

Reply:
Added separate DELETE api - https://github.com/opensearch-project/OpenSearch/pull/4400